Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/ExamGroupsController.cs

```diff
@@ -0,0 +1,286 @@
+﻿namespace OJS.Web.Areas.Administration.Controllers
+{
+    using System.Collections;
+    using System.Data.Entity;
+    using System.Linq;
+    using System.Web.Mvc;
+
+    using Kendo.Mvc.Extensions;
+    using Kendo.Mvc.UI;
+
+    using OJS.Data;
+    using OJS.Data.Models;
+    using OJS.Services.Data.Contests;
+    using OJS.Services.Data.ExamGroups;
+    using OJS.Services.Data.Users;
+    using OJS.Web.Areas.Administration.Controllers.Common;
+    using OJS.Web.Common.Extensions;
+
+    using DetailModelType = OJS.Web.Areas.Administration.ViewModels.User.UserProfileSimpleAdministrationViewModel;
+    using GeneralResource = Resources.Areas.Administration.AdministrationGeneral;
+    using Resource = Resources.Areas.Administration.ExamGroups.ExamGroupsController;
+    using ViewModelType = OJS.Web.Areas.Administration.ViewModels.ExamGroups.ExamGroupAdministrationViewModel;
+
+    public class ExamGroupsController : LecturerBaseGridController
+    {
+        private const int DefaultUsersTakeCount = 20;
+        private const int DefaultContestsToTake = 15;
+
+        private readonly IExamGroupsDataService examGroupsData;
+        private readonly IUsersDataService usersData;
+        private readonly IContestsDataService contestsData;
+
+        public ExamGroupsController(
+            IOjsData data,
+            IExamGroupsDataService examGroupsData,
+            IUsersDataService usersData,
+            IContestsDataService contestsData)
+            : base(data)
+        {
+            this.examGroupsData = examGroupsData;
+            this.usersData = usersData;
+            this.contestsData = contestsData;
+        }
+
+        public ActionResult Index() => this.View();
+
+        public override IEnumerable GetData()
+        {
+            var examGroups = this.examGroupsData.GetAll();
+
+            if (this.User.IsLecturer())
+            {
+                examGroups = examGroups.Where(eg => eg.Contest == null ||
+                    (eg.Contest.Lecturers.Any(l => l.LecturerId == this.UserProfile.Id) ||
+                        eg.Contest.Category.Lecturers.Any(l => l.LecturerId == this.UserProfile.Id)));
+            }
+
+            return examGroups.Select(ViewModelType.FromExamGroup);
+        }
+
+        public override object GetById(object id) => this.GetByIdAsNoTracking((int)id);
+
+        [HttpPost]
+        public ActionResult Create([DataSourceRequest]DataSourceRequest request, ViewModelType model)
+        {
+            var contestId = model.ContestId.HasValue
+                ? this.contestsData.GetByIdQuery(model.ContestId.Value).Select(c => c.Id).FirstOrDefault()
+                : default(int);
+
+            if (contestId != default(int))
```

Comment:
Invert this check. Starting from top it should look something like:
`If !model.ContestId.HasValue` -> error
`contestId = ...`